### PR TITLE
Panic handling

### PR DESCRIPTION
**Pull request by jmank88**
- Restore `WrapRecover` (removed in #5384), but adjust uses to the fix from 1.0.1 (#5395).

### CODE REVIEW

File: core/services/pipeline/runner.go [hunk 1]
```diff
@@ -274,14 +276,24 @@ func (r *runner) run(
...
+		}, func(err interface{}) {
+			t := time.Now()
+			scheduler.report(reportCtx, TaskRunResult{
+				ID:         uuid.NewV4(),
+				Task:       taskRun.task,
+				Result:     Result{Error: ErrRunPanicked{err}},
+				FinishedAt: null.TimeFrom(t),
+				CreatedAt:  t, // TODO: more accurate start time
+			})
+		})
```
Comment (jmank88) [inline core/services/pipeline/runner.go:296]: @archseer Does this effectively reapply the recovery removed here?
https://github.com/smartcontractkit/chainlink/pull/5384/files?diff=unified&w=0#diff-e219e9df67a3181e6e8f0b4806f90b582df4b8d4dc8d3ec85ea132f5cf2dfe9fL273-L286

File: core/services/offchainreporting/run_saver.go [hunk 2]
```diff
@@ -32,12 +33,14 @@ func (r *RunResultSaver) Start() error {
 			for {
 				select {
 				case run := <-r.runResults:
-					r.logger.Infow("RunSaver: saving job run", "run", run)
-					// We do not want save successful TaskRuns as OCR runs very frequently so a lot of records
-					// are produced and the successful TaskRuns do not provide value.
-					if err := r.pipelineRunner.InsertFinishedRun(&run, false); err != nil {
-						r.logger.Errorw("error inserting finished results", "err", err)
-					}
+					recovery.WrapRecover(r.logger, func() {
```
Comment (samsondav) [inline core/services/offchainreporting/run_saver.go:36]: This is so simple and doesnt process any outside input I think its reasonable to not wrap here

**Review by samsondav — CHANGES_REQUESTED** [review]
This adds it back in different places than it used to be, is it more targeted now?

File: core/services/offchainreporting/run_saver.go (hunk 2 above)
Comment (jmank88) [reply core/services/offchainreporting/run_saver.go]: Done

**Review by samsondav — APPROVED** [review]

**Pull request merged**